modules/creating-cluster-with-aws-kms-key.adoc

```diff
@@ -0,0 +1,211 @@
+// Module included in the following assemblies:
+//
+// * monitoring/enabling-monitoring-for-user-defined-projects.adoc
+
+:_content-type: PROCEDURE
+[id="creating-cluster-with-aws-kms-key"]
+= Creating a ROSA cluster using a custom AWS KMS key
+
+You can create a {product-title} (ROSA) cluster with a customer-provided KMS key that is used to encrypt both node root volumes and persistent volumes for mounted EBS claims.
+
+.Procedure
+
+. Create a custom AWS customer-managed KMS key by running the following command:
++
+[source,terminal]
+----
+$ KMS_ARN=$(aws kms create-key --region $AWS_REGION --description 'Custom ROSA Encryption Key' --tags TagKey=red-hat,TagValue=true --query KeyMetadata.Arn --output text)
+----
++
+This command saves the Amazon Resource Name (ARN) output of this custom key for further steps.
++
+[NOTE]
+====
+Customers provide the `--tags TagKey=red-hat,TagValue=true` that are required for a customer KMS key.
+====
+
+
+. Verify the KMS key has been created by running the following command:
++
+[source,terminal]
+----
+$ echo $KMS_ARN
+----
+
+. Set your AWS account ID to an environment variable.
++
+[source,terminal]
+----
+$ AWS_ACCOUNT_ID=<aws-account>
+----
+
+. Generate the necessary key policy to allow the ROSA STS roles to access the custom KMS key. Use the following command to populate a sample policy, or create your own.
+
++
+[source,terminal]
+----
+$ EBS_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-openshift-cluster-csi-drivers-ebs-cloud-credential" | cut -c -64) 
+KUBE_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-kube-controller-manager" | cut -c -64)
+KMS_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-kms-provider" | cut -c -64)
+CAPA_ROLE=$(echo "${OPERATOR_ROLES_PREFIX}-kube-system-capa-controller-manager" | cut -c -64) <1>
+----
++
+[source,terminal]
+----
+cat << EOF > rosa-key-policy.json
+{
+  "Version": "2012-10-17",
+  "Id": "key-rosa-policy-1",
+  "Statement": [
+	{
+  	"Sid": "Enable IAM User Permissions",
+  	"Effect": "Allow",
+  	"Principal": {
+    	"AWS": "arn:aws:iam::${AWS_ACCOUNT_ID}:root"
+  	},
+  	"Action": "kms:*",
+  	"Resource": "*"
+	},
+	{
+  	"Sid": "Allow ROSA use of the key",
+  	"Effect": "Allow",
+  	"Principal": {
+    	"AWS": [
+      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Support-Role",
+      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Installer-Role",
+      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-Worker-Role",
+      	"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-ControlPlane-Role",
```

Comment:
The account role arn for HCP should like this and there is no 'ControlPlane-Role'
```
"arn:aws:iam::<aws_account_id>:role/${ACCOUNT_ROLES_PREFIX}-HCP-ROSA-Installer-Role"
"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-HCP-ROSA-Support-Role",
"arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ACCOUNT_ROLES_PREFIX}-HCP-ROSA-Worker-Role",
```

Reply:
this has been addressed in previous update/comment